server/src/main/java/org/apache/iotdb/db/query/dataset/DeviceIterateDataSet.java

```diff
@@ -207,7 +211,7 @@ protected boolean hasNextWithoutConstraint() throws IOException {
             currentDataSet = queryRouter.fill(fillQueryPlan, context);
             break;
           case QUERY:
-            QueryPlan queryPlan = new QueryPlan();
+            RawDataQueryPlan queryPlan = new RawDataQueryPlan();
```

Comment:
Don't you save three plans in AlignByDevicePlan? They could be reused

Reply:
Good Idea. Fixed.